crates/primitives/src/prune/checkpoint.rs

```diff
@@ -0,0 +1,36 @@
+use crate::{prune::PruneMode, BlockNumber};
+use reth_codecs::{main_codec, Compact};
+
+/// Saves the pruning progress of a stage.
+#[main_codec]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[cfg_attr(test, derive(Default))]
+pub struct PruneCheckpoint {
+    /// Prune mode.
+    prune_mode: PruneMode,
+    /// Highest pruned block number.
+    block_number: BlockNumber,
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::prune::{PruneCheckpoint, PruneMode};
+    use rand::Rng;
+    use reth_codecs::Compact;
+
+    #[test]
+    fn prune_checkpoint_roundtrip() {
```

Comment:
This is unnecessary since deriving `main_codec` automatically adds `derive_arbitrary` which generates these kind of roundtrip tests for a default of 256 cases.  It's mentioned on `derive_arbitratry`, but not on the `main_codec` docs, should probably add.

Reply:
you're right, I see:
```rust
#[cfg(test)]
mod PruneCheckpointTests {
    use super::Compact;

    #[test]
    fn proptest() {
        let mut config = proptest::prelude::ProptestConfig::with_cases(256i32 as u32);
        {
            let mut config = config.__sugar_to_owned();
            ::proptest::sugar::force_no_fork(&mut config);
            {
                config.source_file = Some(file!( ));
                let mut runner = ::proptest::test_runner::TestRunner::new(config);
                let names = stringify!( field );
                match runner.run(
                    &::proptest::strategy::Strategy::prop_map(
                        ::proptest::arbitrary::any::<super::PruneCheckpoint>(),
                        |values| ::proptest::sugar::NamedArguments(names, values)),
                    |::proptest::sugar::NamedArguments(
                         _, field)|
                        {
                            let _: () = {
                                {
                                    let mut buf = (::alloc::vec::Vec::new());
                                    let len = field.clone().to_compact(&mut buf);
                                    let (decoded, _) = super::PruneCheckpoint::from_compact(&buf, len);
                                    assert!(field == decoded);
                                }
                            };
                            Ok(())
                        })
                {
                    Ok(_) => (),
                    Err(e) => panic!("{}\n{}", e, runner),
                }
            }
        }
    }
}
```

removed